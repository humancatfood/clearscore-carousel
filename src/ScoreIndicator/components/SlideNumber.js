import React from 'react';

import { css, cx } from 'emotion';



export const baseStyle = css({
  margin: 5,
  color: '#a9cdd6',
  fontSize: '4rem',
  fontWeight: 'lighter',
});


const SlideNumber = ({ children, className='', ...props }) => (
  <p
    className={cx(baseStyle, className)}
    {...props}
  >
    { children }
  </p>
);

export default SlideNumber;
