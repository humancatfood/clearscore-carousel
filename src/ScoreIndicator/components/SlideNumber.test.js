import React from 'react';
import TestRenderer from 'react-test-renderer';

import SlideNumber, { baseStyle } from './SlideNumber';



describe('SlideNumber', () => {

  let element;

  it('should render', () => {
    element = TestRenderer.create(<SlideNumber />);
    expect(element).toMatchSnapshot();
  });

  it('should accept & pass through props', () => {

    element = TestRenderer.create(<SlideNumber id="blarb"/>);
    expect(element.toJSON().props.id).toBe('blarb');

    element = TestRenderer.create(<SlideNumber className="test-class"/>);
    expect(element.toJSON().props.className).toBe(`${baseStyle} test-class`);

    element = TestRenderer.create(<SlideNumber style={{ backgroundColor: 'blue' }}/>);
    expect(element.toJSON().props.style.backgroundColor).toBe('blue');

  });

});
