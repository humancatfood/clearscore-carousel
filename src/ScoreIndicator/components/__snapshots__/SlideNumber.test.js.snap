// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideNumber should render 1`] = `
<p
  className="css-1r4s34n"
/>
`;
